apps/web/vibes/soul/primitives/skeleton/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { clsx } from 'clsx';
+
+type Props = {
+  parentClassName?: string;
+  childClassName?: string;
+  rounded?: 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'full';
+} & (
+  | { type: 'text'; characterCount: number; aspectRatio?: never }
+  | { type: 'image'; aspectRatio: '5:6' | '3:4' | '1:1'; characterCount?: never }
+  | { type?: 'box' | 'circle'; characterCount?: never; aspectRatio?: never }
+);
+
+/**
+ * This component supports various CSS variables for theming. Here's a comprehensive list, along
+ * with their default values:
+ *
+ * ```css
+ * :root {
+ *   --skeleton-background: hsl(var(--contrast-100));
+ * }
+ * ```
+ */
+
+export function Skeleton({
+  parentClassName,
+  childClassName,
+  type = 'text',
+  rounded,
+  characterCount = 10,
+  aspectRatio = '5:6',
+}: Props) {
+  if (type === 'box') {
+    return (
+      <div className={clsx('flex items-center', parentClassName)}>
+        <div
+          className={clsx(
+            'bg-[var(--skeleton-background,hsl(var(--contrast-100)))]',
```

Comment:
If there's no `--skeleton-text` or `skeleton-foreground` variable we can just call this `--skeleton`.